### DISCUSSION TITLE
How to properly disable the camera manipulators?

### DISCUSSION THREAD

**Discussion by ghost** [google/filament#4325] [Q&A] [answered]
Continuing on the theme of [the issue I opened](https://github.com/google/filament/issues/4318), how can I properly disable the camera manipulators that are built into the samples (I'm starting from the gltf_viewer file)?

I'm trying to add functionality to move the camera to specified positions based on keyboard input. Right now I have the camera changing to the specified positions but it only shows that for a single frame then jumps back to the one forced by the existing manipulators/setup. Commenting out the code related to creating the manipulators causes the original camera position and position it jumps back to but doesn't fix the core issue. How can I do this?

**Comment by prideout** [comment]
Not all samples use the camutils manipulator, for example the `animation.cpp` and `depthtesting.cpp` samples have their own fixed-position cameras.

**Comment by ghost** [comment, marked as answer]
This was answered by Ben in the original thread: Comment out the following line in the FilamentApp.cpp (I also commented out the Manipulator-related code):

```
window->mMainCameraMan->getLookAt(&eye, &center, &up);
```

**Reply by ghost** [reply]
For future readers, once the camera updates were disabled as stated above, the only thing needed to change the camera position is the following:

```
app.mainCamera->lookAt(cameraPosition, pointToLookAt);
```

Where `cameraPosition` and `pointToLookAt` are both `float3`s.
